src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClient.java

```diff
@@ -101,9 +102,10 @@ public StashApiClient(
     } catch (IOException e) {
       logger.log(Level.WARNING, "invalid pull request response.", e);
     }
-    return Collections.emptyList();
+    return null;
```

Comment:
I would rather not use null as special value marker. If we need the calling code to recognize that an error happen why not throw an exception?

Reply:
Good point. Using custom exceptions would improve the code. I started working on it. It would be a bigger change because it would involve changes to the exception handling and logging.

Reply:
I think for now we could just throw some generic exceptions, that would still be better than returning null IMO